winappdbg/process.py

```diff
@@ -1936,11 +1936,18 @@ def read_string(self, lpBaseAddress, nChars, fUnicode = False):
         """
         szString = ''
         if nChars == None:
-            tmp = ''
-            while tmp != '\0':
-                szString += tmp
-                tmp = self.read(lpBaseAddress, 1)
-                lpBaseAddress += 1
+            memory = []
+            for mbi in self.get_memory_map(lpBaseAddress, lpBaseAddress):
+                if mbi.State == win32.MEM_COMMIT and \
+                                            not mbi.Protect & win32.PAGE_GUARD:
+                    memory.append( (mbi.BaseAddress, mbi.RegionSize) )
+            lpBasePage, sizePage = memory[0]

```

Comment:
Wouldn't this cause an IndexError if lpBaseAddress has a bad address? I'd rather have the right type of exception...

Reply:
I'm also thinking if a string is between two pages this would truncate the strings

Reply:
You're right. 
What about the new one?